infra/resources.tf

```diff
@@ -0,0 +1,158 @@
+# Resources to construct on AWS
+
+# Permissions
+data "aws_iam_policy_document" "deploy-user" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "iam:PassRole",
+      "iam:GetRole*",
+      "iam:CreateRole",
+      "iam:PutRolePolicy",
+      "iam:DeleteRolePolicy",
+      "iam:DeleteRole",
+      "iam:AttachRolePolicy"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "cloudformation:CreateStack",
+      "cloudformation:DescribeStacks",
+      "cloudformation:DescribeStackEvents",
+      "cloudformation:DescribeStackResource",
+      "cloudformation:DescribeStackResources",
+      "cloudformation:DeleteStack",
+      "cloudformation:UpdateStack",
+      "cloudformation:ListStackResources"
+    ]
+    resources = [
+      "arn:aws:cloudformation:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:stack/${var.service}*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "cloudformation:ValidateTemplate"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:*"
+    ]
+    resources = [
+      "arn:aws:s3:::${var.deploy_bucket}",
+      "arn:aws:s3:::${var.deploy_bucket}/*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:ListAllMyBuckets",
+      "s3:CreateBucket"
```

Comment:
LGTM. Do we also need put permission here?

Reply:
Which permission?

Reply:
"s3:PutObject"

I'm unsure of the moving pieces here though.